pkg/operation/botanist/component/resourceconfig.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/controllerutils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Class is a type alias for describing the class of a resource.
+type Class uint8
+
+const (
+	// Runtime means that the resource belongs to the runtime cluster (seed).
+	Runtime Class = iota
```

Comment:
Great idea! I think it makes sense to adopt it for `resourcemanager` when we touch it with some larger enhancements next time. I think this package would benefit from it. Do you also have other suggestions where this could be helpful?